### PR TITLE
[Enhancement] add check if vol has available space before create data partition

### PR DESCRIPTION
**Pull request by M1eyu2018**
Add check if vol has available space before create data partition
so that creating new data partition manually or automatically is not allowed when vol usage reaches its capacity.

### CODE REVIEW

File: master/cluster.go [hunk 1]
```diff
@@ -883,6 +883,9 @@ func (c *Cluster) batchCreatePreLoadDataPartition(vol *Vol, preload *DataPartiti
 }
 
 func (c *Cluster) batchCreateDataPartition(vol *Vol, reqCount int) (err error) {
+	if _, err = vol.needCreateDataPartition(); err != nil {
+		return
```
Comment (baijiaruo) [inline master/cluster.go:887]: It is recommended to add a warn alarm log here and print out err

File: master/vol.go [hunk 2]
```diff
@@ -569,31 +569,35 @@ func (vol *Vol) checkAutoDataPartitionCreation(c *Cluster) {
 	}
 }
 
-func (vol *Vol) needCreateDataPartition() bool {
+func (vol *Vol) needCreateDataPartition() (err error) {
```
[the remark this answers is not in the document]
Comment (M1eyu2018) [reply master/vol.go]: Thanks for your review.

File: master/cluster.go (hunk 1 above)
Comment (M1eyu2018) [reply master/cluster.go:888]: Thanks for your review.

**Comment by M1eyu2018** [comment]
hi @sejust, Conflict has been fixed, please review again

**Review by baijiaruo — DISMISSED** [review]

**Review by true1064 — DISMISSED** [review]

**Review by leonrayang — DISMISSED** [review]

**Review by leonrayang — APPROVED** [review]

**Review by leonrayang — APPROVED** [review]

**Pull request merged**